apps/sms/js/navigation.js

```diff
@@ -116,6 +116,17 @@ var Navigation = {
   },
 
   /**
+   * Ensures that current panel is correctly set, if no current panel set and
+   * we're not in the process of transitioning then set default panel as current
+   * one.
+   */
+  ensureCurrentPanel: function() {
```

Comment:
> Initially I put this code into ActivityHandler, but it uses only Navigation internal data, so I moved it to Navigation.js.

I think it's good in Navigation.

For the name, I don't really have a strong opinion.

In my mind, long-term, I think Navigation could handle the initial loading: given a hash, it would lazy load the right JS file (which in turn would lazy load all dependencies), and then call toPanel after the right JS file has returned something (or not) in a promise. So... this name is as good as any other name.